test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildWithATargetPlatform.cs

```diff
@@ -90,5 +92,82 @@ public void It_fails_on_unsupported_os()
                 .And
                 .HaveStdOutContaining("NETSDK1139");
         }
+
+        [Fact]
+        public void It_fails_if_targetplatformversion_is_constant_only()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "It_fails_if_targetplatformversion_is_constant_only",
+                TargetFrameworks = ToolsetInfo.CurrentTargetFramework,
+            };
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+
+            string DirectoryBuildTargetsContent = $@"
+<Project>
+  <ItemGroup>
+    <SdkSupportedTargetPlatformVersion Include=""111.0"" DefineConstantsOnly=""true"" />
+    <SdkSupportedTargetPlatformVersion Include=""222.0"" />
+  </ItemGroup>
+  <PropertyGroup>
+    <TargetPlatformVersion>111.0</TargetPlatformVersion>
+    <TargetPlatformIdentifier>ios</TargetPlatformIdentifier>
+    <TargetPlatformSupported>true</TargetPlatformSupported>
+  </PropertyGroup>
+</Project>
+";
+
+            File.WriteAllText(Path.Combine(testAsset.TestRoot, "Directory.Build.targets"), DirectoryBuildTargetsContent);
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand.Execute()
+                .Should()
+                .Fail()
+                .And
+                .HaveStdOutContaining("NETSDK1140")
+                .And
+                .HaveStdOutContaining(string.Format(Strings.InvalidTargetPlatformVersion, "111.0", "ios", "222.0").Split ('\n', '\r') [0])
+                .And
+                .HaveStdOutContaining("222.0");
+        }
+
+        [Fact]
+        public void It_fails_if_targetplatformversion_is_invalid()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "It_fails_if_targetplatformversion_is_invalid",
+                TargetFrameworks = ToolsetInfo.CurrentTargetFramework,
+            };
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+
+            string DirectoryBuildTargetsContent = $@"
+<Project>
+  <ItemGroup>
+    <SdkSupportedTargetPlatformVersion Include=""222.0"" />
+  </ItemGroup>
+  <PropertyGroup>
+    <TargetPlatformVersion>111.0</TargetPlatformVersion>
+    <TargetPlatformIdentifier>ios</TargetPlatformIdentifier>
+    <TargetPlatformSupported>true</TargetPlatformSupported>
+  </PropertyGroup>
+</Project>
+";
+
+            File.WriteAllText(Path.Combine(testAsset.TestRoot, "Directory.Build.targets"), DirectoryBuildTargetsContent);
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand.Execute()
+                .Should()
+                .Fail()
+                .And
+                .HaveStdOutContaining("NETSDK1140")
+                .And
+                .HaveStdOutContaining("111.0 is not a valid TargetPlatformVersion")
```

Comment:
It would be good to make the string formatting change from the previous test here too.

Reply:
Fixed, forgot I did the same thing twice.